File: ios/Plugin/ApplicationContextPlugin.swift
import Foundation
import Capacitor
import WatchConnectivity

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(ApplicationContextPlugin)
public class ApplicationContextPlugin: CAPPlugin {
    private let implementation = ApplicationContext()

    @objc func updateApplicationContext(_ call: CAPPluginCall) {
        do {
            try WCSession.default.updateApplicationContext(call.dictionaryRepresentation as! [String : Any])
            print("Called updateApplicationContext successfully.")
        } catch let error {
            print("Error updating application context: \(error.localizedDescription). Have you set up the watch delegates?")
        }

        call.resolve()
    }
}

File: android/src/main/java/com/mycompany/plugins/example/ApplicationContextPlugin.java
package com.mycompany.plugins.example;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "ApplicationContext")
public class ApplicationContextPlugin extends Plugin {

    private ApplicationContext implementation = new ApplicationContext();

    @PluginMethod
    public void updateApplicationContext(PluginCall call) {
        call.resolve();
    }
}
